### PR TITLE
Rework of feature flagging on Add Account flows

### PR DESCRIPTION
**Pull request by lvndry**
### 📝 Description

Rework add account feature flagging in manager app.
We fetch all existing flags and check if any currency app is impacted by the feature flag

### ❓ Context

- **Impacted projects**: `LLD` `LLC`
- **Linked resource(s)**:
  - https://ledgerhq.atlassian.net/browse/LIVE-7344
  - https://ledgerhq.atlassian.net/browse/LIVE-5099
  - https://ledgerhq.atlassian.net/browse/LIVE-5100

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (lambertkevin) [inline apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:38]: Please add a `useCallback`

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (lambertkevin) [inline apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:46]: Please add a `useCallback`

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (lambertkevin) [inline apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:64]: I know it's not part of your modification, but add it here as well please 🙏

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (lambertkevin) [inline apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:52]: Would it work with `feature_currency_` instead ? To me, the more specific we are, the better.

(Example that I have in mind, if we decide to do the remote config with a `config_currency_XXX` 🤷‍♂️ )

**Review by lambertkevin — CHANGES_REQUESTED** [review]
Only some remarks regarding the React memoization part, wait for another review anyway from someone who knows a little bit more about feature flags than me 😝

**Comment by gre** [comment]
👋  we just merged https://github.com/LedgerHQ/ledger-live/pull/3359 which greatly improved the TypeScript checks on Ledger Live Desktop projects (components, modals, screens) and effectively make it type check on the CI. Could you rebase with it and ensure there is no issue on `pnpm desktop typecheck`? Thanks and don't hesitate to ask for any TS help.

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (lvndry) [reply apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:46]: Yes I could use only only `getAllFlags`, however we will just loop over every feature flags in the manager app even though they are not currency feature flags

**Review by haammar-ledger — APPROVED** [review]
LGTM

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx
Comment (haammar-ledger) [reply apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx]: You're right, but I don't think there will ever be so many feature flags that there is a real perf impact.

File: apps/ledger-live-desktop/src/renderer/screens/manager/AppsList/Item.tsx
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/screens/manager/AppsList/Item.tsx:87]: ⚠️  i'm not familiar enough with firebase framework but are we sure this is not costly?
actually i'm really worried about the performance of this. 
- getAllFlags is implementing with a bunch of JSON.parse and it's fetching ALL flags where you instead want to only ask "is my currency concerned" (and you don't care about parsing the others).
- this is not secured by a useMemo, so this will run on each single re-render
- this will be running for each item in the AppsList then!🤔  

so we have AppsCount x FlagsCount number of JSON parse on each re-render.

**Review by gre — CHANGES_REQUESTED** [review]
on the algorithmic code review of this, it's perf nogo to me.

also do we have ways to cover this code by test?

File: apps/ledger-live-desktop/src/renderer/screens/manager/AppsList/Item.tsx
Comment (lvndry) [reply apps/ledger-live-desktop/src/renderer/screens/manager/AppsList/Item.tsx:87]: We can't add query parameters to only fetch currency flags on firebase. These are the only available methods: https://firebase.google.com/docs/reference/js/v8/firebase.remoteconfig.RemoteConfig#methods

My initial solution was to only parse the currency flags but an other PR comment recommended to do otherwise (https://github.com/LedgerHQ/ledger-live/pull/3320#discussion_r1184696787)

Indeed I should pass the flags to the items and not do the fetch from there I'll change that

**Review by gre — APPROVED** [review]
overall i'm ok with the way this remains contained but i have some reserve on the approach to `JSON.parse` a lot of things

**Pull request merged**
